Competition2018/src/main/java/competition/subsystems/autonomous/AutonomousDecisionSystem.java

```diff
@@ -7,107 +7,156 @@
 import com.google.inject.Inject;
 import com.google.inject.Singleton;
 
-import openrio.powerup.MatchData;
 import openrio.powerup.MatchData.GameFeature;
 import openrio.powerup.MatchData.OwnedSide;
 import xbot.common.command.BaseSubsystem;
 import xbot.common.math.ContiguousHeading;
 import xbot.common.math.FieldPose;
 import xbot.common.math.XYPair;
+import xbot.common.properties.BooleanProperty;
+import xbot.common.properties.XPropertyManager;
 
 @Singleton
 public class AutonomousDecisionSystem extends BaseSubsystem {
 
-    GameDataSource gameData;
+    final GameDataSource gameData;
+    final BooleanProperty robotOnRight;
     
     @Inject
-    public AutonomousDecisionSystem(GameDataSource gameData) {
+    public AutonomousDecisionSystem(GameDataSource gameData, XPropertyManager propMan) {
```

Comment:
The one auto thing we leave to a property for now is "is the robot starting on the right" to avoid a total explosion in commands.